tests/test_fastapi.py

```diff
@@ -0,0 +1,71 @@
+import uuid
+from typing import Tuple
+
+import sqlalchemy as sa
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+
+from dbos_transact.context import SetWorkflowUUID
+from dbos_transact.dbos import DBOS
+
+
+def test_simple_endpoint(dbos_fastapi: Tuple[DBOS, FastAPI]) -> None:
+    dbos, app = dbos_fastapi
+    client = TestClient(app)
+
+    @app.get("/{var1}/{var2}")
+    @dbos.workflow()
+    def test_workflow(var1: str, var2: str) -> str:
+        assert DBOS.request is not None
+        res1 = test_transaction(var1)
+        res2 = test_communicator(var2)
+        return res1 + res2
+
+    @dbos.transaction()
+    def test_transaction(var: str) -> str:
+        rows = DBOS.sql_session.execute(sa.text("SELECT 1")).fetchall()
+        return var + str(rows[0][0])
+
+    @dbos.communicator()
+    def test_communicator(var: str) -> str:
+        return var
+
+    response = client.get("/bob/bob")
+    assert response.status_code == 200
+    assert response.text == '"bob1bob"'
+
+
+def test_endpoint_recovery(dbos_fastapi: Tuple[DBOS, FastAPI]) -> None:
+    dbos, app = dbos_fastapi
+    client = TestClient(app)
+
+    @app.get("/{var1}")
```

Comment:
Maybe also test if we can invoke a workflow from a normal FastAPI endpoint? Just to make sure contexts are passed in correctly.